common/test/acceptance/tests/lms/test_lms_problems.py

```diff
@@ -102,7 +102,61 @@ def test_clarification(self):
         self.assertNotIn('strong', tooltip_text)
 
 
-class ProblemExtendedHintTest(ProblemsTest, EventsTestMixin):
+class ProblemHintTest(ProblemsTest, EventsTestMixin):
+    """
+    Base test class for problem hint tests.
+    """
+    def verify_check_hint(self, answer, answer_text, expected_events):
+        """
+        Verify clicking Check shows the extended hint in the problem message.
+        """
+        self.courseware_page.visit()
+        problem_page = ProblemPage(self.browser)
+        self.assertEqual(problem_page.problem_text[0], u'question text')
+        problem_page.fill_answer(answer)
+        problem_page.click_submit()
+        self.assertEqual(problem_page.message_text, answer_text)
+        # Check for corresponding tracking event
+        actual_events = self.wait_for_events(
+            event_filter={'event_type': 'edx.problem.hint.feedback_displayed'},
+            number_of_matches=1
+        )
+        self.assert_events_match(expected_events, actual_events)
+
+    def verify_demand_hints(self, first_hint, second_hint, expected_events):
+        """
+        Test clicking through the demand hints and verify the events sent.
+        """
+        self.courseware_page.visit()
+        problem_page = ProblemPage(self.browser)
+
+        # The single visible Hint button should be enabled.
```

Comment:
Note to reviewers: the guts of this test method changed to reflect changes in hint behavior.